packages/workspace/src/browser/workspace-commands.ts

```diff
@@ -447,15 +447,29 @@ export class WorkspaceRootUriAwareCommandHandler extends UriAwareCommandHandler<
         super(selectionService, handler);
     }
 
+    isVisible(): boolean {
+        return super.isVisible();
+    }
+
+    isEnabled(): boolean {
+       return super.isEnabled();
+    }
+
     protected getUri(): URI | undefined {
         const uri = super.getUri();
-        if (this.workspaceService.isMultiRootWorkspaceEnabled) {
-            return uri;
-        }
+        // If the URI is available, return it immediately.
         if (uri) {
             return uri;
         }
-        const root = this.workspaceService.tryGetRoots()[0];
+        // If a multi-root workspace is currently opened with at least one root,
```

Comment:
Why not return first root or `undefined`?

Code below will break if there mutli-root workspaces is opened without any roots. Please make sure to not use `this.workspaceService.workspace`, but only `roots`.

Reply:
so just `this.workspaceService.tryGetRoots()[0]`;